tools/sslsniff.py

```diff
@@ -48,41 +56,69 @@
                     help=argparse.SUPPRESS)
 parser.add_argument("--hexdump", action="store_true", dest="hexdump",
                     help="show data as hexdump instead of trying to decode it as UTF-8")
+parser.add_argument('--max-buffer-size', type=int, default=8192,
+                    help='Size of captured buffer')
 args = parser.parse_args()
 
 
 prog = """
 #include <linux/ptrace.h>
 #include <linux/sched.h>        /* For TASK_COMM_LEN */
 
+#define MAX_BUF_SIZE __MAX_BUF_SIZE__
+
 struct probe_SSL_data_t {
         u64 timestamp_ns;
         u32 pid;
-        char comm[TASK_COMM_LEN];
-        char v0[464];
+        u32 tid;
+        u32 uid;
         u32 len;
+        int buf_filled;
+        char comm[TASK_COMM_LEN];
+        u8 buf[MAX_BUF_SIZE];
 };
 
+#define BASE_EVENT_SIZE ((size_t)(&((struct probe_SSL_data_t*)0)->buf))
+#define EVENT_SIZE(X) (BASE_EVENT_SIZE + ((size_t)(X)))
+
+
+BPF_PERCPU_ARRAY(ssl_data, struct probe_SSL_data_t, 1);
```

Comment:
percpu array is supported since 4.6. I think we are fine here as ubuntu 16.04 (4.4 kernel) is EOL now.

Reply:
I've tested it on `4.19` and `4.15` kernels (`Ubuntu 18.04`), so it's fine in this regard.